rustls/src/crypto/cipher.rs

```diff
@@ -16,9 +15,9 @@ pub trait Tls13AeadAlgorithm: Send + Sync {
     /// The length of key in bytes required by `encrypter()` and `decrypter()`.
     fn key_len(&self) -> usize;
 
-    #[cfg(feature = "secret_extraction")]
-    /// Convert the key material from `key`/`iv`, into a `ConnectionTrafficSecrets` item.
-    fn extract_keys(&self, key: AeadKey, iv: Iv) -> ConnectionTrafficSecrets;
+    /// Convert the key material from `key`/`iv`, into a `ConnectionTrafficSecrets` item. May
+    /// return an `Error` if the AEAD algorithm is not a supported variant of `ConnectionTrafficSecrets`.
+    fn extract_keys(&self, key: AeadKey, iv: Iv) -> Result<ConnectionTrafficSecrets, Error>;
```

Comment:
It might make more sense to avoid using `rustls::Error` here in place of a more specific "Unsupported Operation" error. WDYT?

Reply:
Yeah, I think having a specific error probably makes sense here.

Reply:
Switched to a specific error.